### ISSUE TITLE
iw_scan: mismatch RNS capabilities with HT capabilities

### ISSUE THREAD

**Issue by cbeaujoin-stellar** [kellyjonbrazil/jc#610]
**Command output:**
```
BSS xx:xx:xx:xx:3e:41(on wlan0-1)
	last seen: 4206.107s [boottime]
	TSF: 608162896731 usec (7d, 00:56:02)
	freq: 2412
	beacon interval: 100 TUs
	capability: ESS (0x1431)
	signal: -54.00 dBm
	last seen: 1410 ms ago
	Information elements from Probe Response frame:
	SSID: Troubleshooting
	RSN:	 * Version: 1
		 * Group cipher: CCMP
		 * Pairwise ciphers: CCMP
		 * Authentication suites: PSK
		 * Capabilities: 16-PTKSA-RC 1-GTKSA-RC (0x000c)
	HT capabilities:
		Capabilities: 0x1ef
			RX LDPC
			HT20/HT40
			SM Power Save disabled
			RX HT20 SGI
			RX HT40 SGI
			TX STBC
			RX STBC 1-stream
			Max AMSDU length: 3839 bytes
			No DSSS/CCK HT40
		Maximum RX AMPDU length 65535 bytes (exponent: 0x003)
		Minimum RX AMPDU time spacing: No restriction (0x00)
		HT RX MCS rate indexes supported: 0-15
		HT TX MCS rate indexes are undefined
	HT operation:
		 * primary channel: 1
		 * secondary channel offset: no secondary
		 * STA channel width: 20 MHz
	VHT capabilities:
		VHT Capabilities (0x33800192):
			Max MPDU length: 11454
			Supported Channel Width: neither 160 nor 80 80
			RX LDPC
			TX STBC
			RX antenna pattern consistency
			TX antenna pattern consistency
		VHT RX MCS set:
			1 streams: MCS 0-9
			2 streams: MCS 0-9
			3 streams: not supported
			4 streams: not supported
			5 streams: not supported
			6 streams: not supported
			7 streams: not supported
			8 streams: not supported
		VHT RX highest supported: 780 Mbps
		VHT TX MCS set:
			1 streams: MCS 0-9
			2 streams: MCS 0-9
			3 streams: not supported
			4 streams: not supported
			5 streams: not supported
			6 streams: not supported
			7 streams: not supported
			8 streams: not supported
		VHT TX highest supported: 780 Mbps
		VHT extended NSS: not supported
	VHT operation:
		 * channel width: 0 (20 or 40 MHz)
		 * center freq segment 1: 0
		 * center freq segment 2: 0
		 * VHT basic MCS set: 0xfffa
	HE capabilities:
		HE MAC Capabilities (0x010102000040):
			 HTC HE Supported
			Minimum Payload size of 128 bytes: 1
			OM Control
			A-MSDU in A-MPDU
		HE PHY Capabilities: (0x06304c090c008008020c00):
			HE40/2.4GHz
			HE40/HE80/5GHz
			Device Class: 1
			LDPC Coding in Payload
			STBC Tx <= 80MHz
			STBC Rx <= 80MHz
			Full Bandwidth UL MU-MIMO
			DCM Max Constellation: 1
			DCM Max Constellation Rx: 1
			Beamformee STS <= 80Mhz: 3
			PPE Threshold Present
			Max NC: 1
			20MHz in 40MHz HE PPDU 2.4GHz
			TX 1024-QAM
			RX 1024-QAM
		HE RX MCS and NSS set <= 80 MHz
			1 streams: MCS 0-11
			2 streams: MCS 0-11
			3 streams: not supported
			4 streams: not supported
			5 streams: not supported
			6 streams: not supported
			7 streams: not supported
			8 streams: not supported
		HE TX MCS and NSS set <= 80 MHz
			1 streams: MCS 0-11
			2 streams: MCS 0-11
			3 streams: not supported
			4 streams: not supported
			5 streams: not supported
			6 streams: not supported
			7 streams: not supported
			8 streams: not supported
		PPE Threshold 0x19 0x1c 0xc7 0x71 
```

**Pretty result:**
```
  {
    "bssid": "xx:xx:xx:xx:3e:41",
    "interface": "wlan0-1",
    "freq": 2412,
    "capability": "ESS (0x1431)",
    "ssid": "Troubleshooting",
    "rsn": "Version: 1",
    "group_cipher": "CCMP",
    "pairwise_ciphers": "CCMP",
    "authentication_suites": "PSK",
    "capabilities": "0x1ef",
    "ht_rx_mcs_rate_indexes_supported": "0-15",
    "primary_channel": 1,
    "secondary_channel_offset": "no secondary",
    "max_mpdu_length": 11454,
    "supported_channel_width": "neither 160 nor 80 80",
    "1_streams": "MCS 0-11",
    "2_streams": "MCS 0-11",
    "3_streams": "not supported",
    "4_streams": "not supported",
    "5_streams": "not supported",
    "6_streams": "not supported",
    "7_streams": "not supported",
    "8_streams": "not supported",
    "vht_extended_nss": "not supported",
    "channel_width": "0 (20 or 40 MHz)",
    "center_freq_segment_1": 0,
    "center_freq_segment_2": 0,
    "vht_basic_mcs_set": "0xfffa",
    "minimum_payload_size_of_128_bytes": 1,
    "he_phy_capabilities": "(0x06304c090c008008020c00):",
    "device_class": 1,
    "dcm_max_constellation": 1,
    "dcm_max_constellation_rx": 1,
    "beamformee_sts_<=_80mhz": 3,
    "max_nc": 1,
    "tsf_usec": 608162896731,
    "sta_channel_width_mhz": 20,
    "beacon_interval_tus": 100,
    "signal_dbm": -54.0,
    "last_seen_ms": 1410,
    "max_amsdu_length_bytes": 3839,
    "minimum_rx_ampdu_time_spacing_usec": "No",
    "vht_rx_highest_supported_mbps": 780,
    "vht_tx_highest_supported_mbps": 780
  }
```

**Raw output:**
```
 {
    "bssid": "80:cc:9c:bf:3e:41",
    "interface": "wlan0-1",
    "last_seen": "1410 ms ago",
    "tsf": "608162896731 usec (7d, 00:56:02)",
    "freq": "2412",
    "beacon_interval": "100 TUs",
    "capability": "ESS (0x1431)",
    "signal": "-54.00 dBm",
    "information_elements_from_probe_response_frame": "",
    "ssid": "Stellar-Troubleshooting",
    "rsn": "* Version: 1",
    "group_cipher": "CCMP",
    "pairwise_ciphers": "CCMP",
    "authentication_suites": "PSK",
    "capabilities": "0x1ef",
    "ht_capabilities": "",
    "max_amsdu_length": "3839 bytes",
    "minimum_rx_ampdu_time_spacing": "No restriction (0x00)",
    "ht_rx_mcs_rate_indexes_supported": "0-15",
    "ht_operation": "",
    "primary_channel": "1",
    "secondary_channel_offset": "no secondary",
    "sta_channel_width": "20 MHz",
    "vht_capabilities": "",
    "vht_capabilities_0x33800192": "",
    "max_mpdu_length": "11454",
    "supported_channel_width": "neither 160 nor 80 80",
    "vht_rx_mcs_set": "",
    "1_streams": "MCS 0-11",
    "2_streams": "MCS 0-11",
    "3_streams": "not supported",
    "4_streams": "not supported",
    "5_streams": "not supported",
    "6_streams": "not supported",
    "7_streams": "not supported",
    "8_streams": "not supported",
    "vht_rx_highest_supported": "780 Mbps",
    "vht_tx_mcs_set": "",
    "vht_tx_highest_supported": "780 Mbps",
    "vht_extended_nss": "not supported",
    "vht_operation": "",
    "channel_width": "0 (20 or 40 MHz)",
    "center_freq_segment_1": "0",
    "center_freq_segment_2": "0",
    "vht_basic_mcs_set": "0xfffa",
    "he_capabilities": "",
    "he_mac_capabilities_0x010102000040": "",
    "minimum_payload_size_of_128_bytes": "1",
    "he_phy_capabilities": "(0x06304c090c008008020c00):",
    "device_class": "1",
    "dcm_max_constellation": "1",
    "dcm_max_constellation_rx": "1",
    "beamformee_sts_<=_80mhz": "3",
    "max_nc": "1"
  }
```

Current:
```
capabilities: `0x1ef`
```

Expected:
```
"rsn_capabilities": "0x000c",
"ht_capabilities": "0x1ef",
```

**Comment by kellyjonbrazil** [comment]
Hi there! This is a pretty horrendous, beta quality (should be alpha) parser. 😃 

I added some code to allow it to grab more fields. It's not pretty and I'm not sure how much more time I'll spend on this parser since so many of the fields are unknown. I hope this helps!

This update will make it into the v1.25.4 release. You can test it in the meantime by copying the parser file to your plugin directory:

https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/iw_scan.py

- Linux/unix: `$HOME/.local/share/jc/jcparsers`
- macOS: `$HOME/Library/Application Support/jc/jcparsers`
- Windows: `$LOCALAPPDATA\jc\jc\jcparsers`

Let me know how it goes!

**Comment by kellyjonbrazil** [comment]
Now available in release v1.25.4
